src/main/java/no/nav/familie/inntektsmelding/imdialog/tjenester/GrunnlagDtoTjeneste.java

```diff
@@ -96,25 +104,58 @@ public InntektsmeldingDialogDto lagArbeidsgiverinitiertDialogDto(PersonIdent fø
             personInfo.fødselsnummer().getIdent(), personInfo.aktørId().getAktørId());
         var organisasjonDto = lagOrganisasjonDto(organisasjonsnummer.orgnr());
         var innmelderDto = lagInnmelderDto(ytelsetype);
-        var inntektDtoer = lagInntekterDto(null, personInfo.aktørId(), førsteFraværsdag, organisasjonsnummer.orgnr());
-        return new InntektsmeldingDialogDto(personDto,
+
+        return new ArbeidsgiverinitiertDialogDto(personDto,
             organisasjonDto,
             innmelderDto,
-            inntektDtoer,
-            førsteFraværsdag,
             KodeverkMapper.mapYtelsetype(ytelsetype),
+            førsteFraværsdag,
             null,
-            KodeverkMapper.mapForespørselStatus(ForespørselStatus.UNDER_BEHANDLING),
-            førsteFraværsdag
-        );
+            finnAnsettelsesperioder(fødselsnummer, organisasjonsnummer, førsteFraværsdag));
+    }
+
+    private List<ArbeidsgiverinitiertDialogDto.AnsettelsePeriodeDto> finnAnsettelsesperioder(PersonIdent fødselsnummer,
+                                                                                             OrganisasjonsnummerDto organisasjonsnummer,
+                                                                                             LocalDate førsteFraværsdag) {
+        return arbeidsforholdTjeneste.hentArbeidsforhold(fødselsnummer, førsteFraværsdag).stream()
+            .filter(arbeidsforhold -> arbeidsforhold.organisasjonsnummer().equals(organisasjonsnummer.orgnr()))
+            .map(arbeidsforhold -> mapAnsettelsePeriode(arbeidsforhold.ansettelsesperiode()))
+            .toList();
+    }
+
+    private ArbeidsgiverinitiertDialogDto.AnsettelsePeriodeDto mapAnsettelsePeriode(AnsettelsesperiodeDto ansettelsesperiode) {
+        if (ansettelsesperiode != null) {
+            return new ArbeidsgiverinitiertDialogDto.AnsettelsePeriodeDto(ansettelsesperiode.periode().fom(),
+                ansettelsesperiode.periode().tom() != null ? ansettelsesperiode.periode().tom() : Tid.TIDENES_ENDE);
+        }
+        return null;
+    }
+
+    private ArbeidsgiverinitiertDialogDto lagArbeidsgiverinitiertDialogDtoForEksisterendeForespørsel(ForespørselEntitet forespørsel, PersonIdent fødselsnummer) {
+        var ytelseType = forespørsel.getYtelseType();
+        var personDto = lagPersonDto(forespørsel.getAktørId(), ytelseType);
+        var organisasjon = forespørsel.getOrganisasjonsnummer();
+        var organisasjonDto = lagOrganisasjonDto(organisasjon);
+        var innmelderDto = lagInnmelderDto(ytelseType);
+        var førsteUttaksdato = forespørsel.getFørsteUttaksdato();
+
+        return new ArbeidsgiverinitiertDialogDto(personDto,
+            organisasjonDto,
+            innmelderDto,
+            KodeverkMapper.mapYtelsetype(ytelseType),
+            førsteUttaksdato,
+            forespørsel.getUuid(),
+            //Todo skal jeg sende opp ansettelsesperioder her, eller kommer front end uansett til å hente dette
```

Comment:
Hmm, hvis det finnes UUID vil jeg redirecte til eksisterende løype og hente "vanlige" opplysninger. Da bruker vi ikke ansettelseperioder (fordi dato ikke kan endres) så har ikke bruk for det.

Kan godt gi tom liste i dette caset for å slippe gjøre kall mot arbeidsforhold

Reply:
Jeg har oppdatert PR slik vi ble enige om i møtet; ha samme dto for arbeidsgiverinitiert og vanlig inntektsmelding